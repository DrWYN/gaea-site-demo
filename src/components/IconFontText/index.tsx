import * as React from 'react';

import iconfont from 'Src/fonts/index';

interface ITextProps {
  icon?: string;
  testID?: string;
  className?: string;
  style?: any;
}

const initialStyles = {
    fontFamily: 'iconfont',
}

const IconFontText = (props: ITextProps) => {
  const { icon, style = {}, testID, ...other } = props;
    return (
      <span data-testid={testID} {...other}
          style={{
          ...initialStyles,
          ...style
        }}
        >
      { icon && iconfont[icon] }
      </span>
    )
}

export default IconFontText;

