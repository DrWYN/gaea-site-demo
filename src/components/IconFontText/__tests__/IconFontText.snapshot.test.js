import React from 'react';
import renderer from 'react-test-renderer';
import IconFontText from '../index';

describe('<IconFontText />', () => {

  it('props', () => {
    const component = renderer.create(
      <IconFontText
        icon={'test'}
        testID={'test'}
        className={'test'}
        style={{}}
      />
    );

    let snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
})
