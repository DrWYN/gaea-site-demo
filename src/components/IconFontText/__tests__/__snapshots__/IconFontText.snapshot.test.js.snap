// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconFontText /> props 1`] = `
<span
  className="test"
  data-testid="test"
  style={
    Object {
      "fontFamily": "iconfont",
    }
  }
/>
`;
